integrations/maven-bloop/src/main/scala/bloop/integrations/maven/MojoImplementation.scala

```diff
@@ -0,0 +1,106 @@
+  private val ScalaMavenGroupArtifact = "net.alchim31.maven:scala-maven-plugin"
+
+  def initializeMojo(project: MavenProject,
+                     session: MavenSession,
+                     mojoExecution: MojoExecution,
+                     mavenPluginManager: MavenPluginManager,
+                     encoding: String): BloopMojo = {
+    val currentConfig = mojoExecution.getConfiguration()
+    val scalaMavenPlugin = Option(project.getBuild().getPluginsAsMap().get(ScalaMavenGroupArtifact))
+      .getOrElse(sys.error(s"The plugin $ScalaMavenGroupArtifact could not be found."))
+    val scalaMavenConfig = scalaMavenPlugin.getConfiguration().asInstanceOf[Xpp3Dom]
+    mojoExecution.setConfiguration(Xpp3Dom.mergeXpp3Dom(currentConfig, scalaMavenConfig))
+    mavenPluginManager
+      .getConfiguredMojo(classOf[Mojo], session, mojoExecution)
+      .asInstanceOf[BloopMojo]
+  }
+
+  private val emptyLauncher = new AppLauncher("", "", Array(), Array())
+  private val testFrameworks: List[List[String]] = List(
+    List("org.scalacheck.ScalaCheckFramework"),
+    List("org.scalatest.tools.Framework", "org.scalatest.tools.ScalaTestFramework"),
+    List("org.specs.runner.SpecsFramework",
+         "org.specs2.runner.Specs2Framework",
+         "org.specs2.runner.SpecsFramework"),
+    List("com.novocode.junit.JUnitFramework")
+  )
+
+  def writeCompileAndTestConfiguration(mojo: BloopMojo, log: Log): Unit = {
+    import scala.collection.JavaConverters._
+    val project = mojo.getProject()
+    val configDir = mojo.getBloopConfigDir.toPath()
+    if (!Files.exists(configDir)) Files.createDirectory(configDir)
+
+    val launcherId = mojo.getLauncher()
+    val launchers = mojo.getLaunchers()
+    val launcher = launchers
+      .find(_.getId == launcherId)
+      .getOrElse {
+        if (launchers.isEmpty)
+          log.info(s"Using empty launcher: no run setup for ${project.getName}.")
+        else if (launcherId.nonEmpty)
+          log.warn(s"Using empty launcher: launcher id '${launcherId}' doesn't exist.")
+        emptyLauncher
+      }
+
+    def writeConfig(sourceDirs0: Seq[File],
+                    classesDir0: File,
+                    classpath0: java.util.List[_],
+                    launcher: AppLauncher,
+                    configuration: String): Unit = {
+      val name = project.getArtifactId()
+      val build = project.getBuild()
+      val baseDirectory = project.getBasedir()
+      val sourceDirs = sourceDirs0.map(_.getCanonicalFile())
+      val classesDir = classesDir0.getCanonicalFile()
+      val classpath = {
+        val cp = classpath0.asScala.toList.asInstanceOf[List[String]].map(new File(_))
+        if (cp.headOption.contains(classesDir)) cp.tail else cp
+      }
+      val dependencies = project.getProjectReferences().asScala.values.map(_.getArtifactId).toList
+      val allScalaJars = mojo.getAllScalaJars()
+      val tmpDir = new File(classesDir, "tmp-bloop")
+      val fork = mojo.getFork()
```

Comment:
It looks like `fork` from `scala-maven-plugin` actually means whether compilation should happen in a forked JVM? My reading is that `run` is always forked, and that for running the tests, it's the responsibility of another plugin. I'd set it to `true` here too.

https://github.com/davidB/scala-maven-plugin/blob/ef3d064eaf9774859ffec1053962f6fc8117d19e/src/main/java/scala_maven/ScalaMojoSupport.java#L286-L291